netbox_onboarding/tests/test_onboard.py

```diff
@@ -182,3 +191,28 @@ def test_ensure_primary_ip_not_exist(self):
         nbk.ensure_primary_ip()
         self.assertIsInstance(nbk.primary_ip, IPAddress)
         self.assertEqual(nbk.primary_ip.interface, nbk.interface)
+
+    @mock.patch("netbox_onboarding.onboard.socket.gethostbyname")
+    def test_check_ip(self, mock_get_hostbyname):
+        """Check DNS to IP address."""
+        # Look up response value
+        mock_get_hostbyname.return_value = "192.0.2.1"
+
+        # Create a Device Keeper object of the device
+        ndk4 = NetdevKeeper(self.onboarding_task4)
+
+        # Check that the IP address is returned
+        self.assertTrue(ndk4.check_ip())
+
+        # Run the check to change the IP address
+        ndk4.check_ip()
+        self.assertEqual(ndk4.ot.ip_address, "192.0.2.1")
+
+    @mock.patch("netbox_onboarding.onboard.socket.gethostbyname")
+    def test_failed_check_ip(self, mock_get_hostbyname):
+        """Check DNS to IP address failing."""
+        # Look up a failed response
+        mock_get_hostbyname.side_effect = gaierror(8)
+        ndk5 = NetdevKeeper(self.onboarding_task5)
+
+        self.assertFalse(ndk5.check_ip())
```

Comment:
It would be good to assert that `ndk5.ot.ip_address` is unchanged after this call.